wordpress-mobile/WordPress-iOS
# [Xcode 12] Update Core Data transformers to use NSSecureUnarchiveFromData

Fixes #15167.

This PR makes some changes to remove the `is using a nil or insecure value transformer` warnings when building with Xcode 12.

- Updates all Transformable Core Data properties that use 'top level' classes to specify the `NSSecureUnarchiveFromData` transformer. This only exists on iOS 12+, so this will generate console warnings on iOS 11. However, if the class isn't found, I'm pretty certain that the system will just fallback to the default behaviour.
- Adds custom value transformers for the two Transformables that don't use top level classes – one is our own Coordinate class, and the other is NSError (which does conform to NSSecureCoding).

**To test**

- Note: I should probably add a unit test to verify the migration before we merge this
- Install a build that uses the 102 version of the data model
- In that build, create a post and in Post Settings assign it a location
- Also, begin uploading some media in the Media library and cause it to error - the easiest way is to disable your internet connection during the upload
- Background the app to ensure everything is saved
- Install a build from this branch, and ensure there are no Core Data errors logged on startup
- Ensure that your post and its location are still correct
- Visit the media library, tap on one of your failed media items and check the error message is shown as the title of the alert controller that appears.

PR submission checklist:
